crates/bevy_window/src/window.rs

```diff
@@ -123,35 +123,30 @@ impl Window {
     }
 
     #[inline]
-    pub fn width(&self) -> u32 {
-        self.width
+    pub fn logical_width(&self) -> f32 {
```

Comment:
I think this should probably still return a u32. Users shouldn't need to decide how to round these values / it should behave like a physical pixel. If for some reason floats are necessary, they can always calculate that themselves using physical_width and scale_factor

Reply:
I would think rounding would be wrong as it will lose precision. Pardon my ignorance, but what the use case for wanting integral coordinates?

Reply:
You are right to say that this is a loss of precision, but pixels are a "discrete" measure. People can (and should) write code assuming that the thing they're rendering to is made out of pixels. Whether they are logical or physical pixels shouldn't matter.

Reply:
One place where usage of height/width as u32 that looks suspicious to me is `CameraProjection::update`. I would think this trait should be updated to received the potentially fractional pixel coordinates too.

Reply:
would it be possible to have a `fn width<X: Pixel>(&self) -> X` ?
like winit does with the [`Pixel`](https://docs.rs/winit/0.23.0/winit/dpi/trait.Pixel.html) trait to let caller choose the precision they want

Reply:
I have a hard time with this one. I agree that physical pixels are discrete, but the same cannot be said for logical pixels.

Reply:
@mockersf  That is certainly possible. From what I understand, winit uses this primarily to decided between physical and logical coordinates which if we followed, would also allow us to remove the physical_XXX versions of these methods.

Reply:
I reverted `width` & `height` to returning `u32` but added back the `logical` versions that return the full precision which is used in a few places in the engine where floats where needed (assuming they would benefit from the full precision)

Reply:
As mentioned in the other comment, I'm open to moving to floats for default width()/height() if we can construct a case where it matters, but lets do that outside of this pr.